src/lib/backend_fapi.c

```diff
@@ -0,0 +1,490 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include "config.h"
+#include "backend_fapi.h"
+#include "emitter.h"
+#include "parser.h"
+#include <tss2/tss2_fapi.h>
+
+FAPI_CONTEXT *fctx = NULL;
+int objectids = 0;
+
+CK_RV backend_fapi_init(void) {
+    if (fctx != NULL) {
+        LOGW("Backend FAPI already initialized.");
+        return CKR_OK;
+    }
+    LOGV("Calling Fapi_Initialize");
+    TSS2_RC rc = Fapi_Initialize(&fctx, NULL);
+    if (rc) {
+        LOGW("Could not initialize FAPI");
+        return CKR_GENERAL_ERROR;
+    }
+    return CKR_OK;
+}
+
+CK_RV backend_fapi_destroy(void) {
+    LOGV("Calling Fapi_Finalize");
+    Fapi_Finalize(&fctx);
+    return CKR_OK;
+}
+
+CK_RV backend_fapi_ctx_new(token *t) {
+    t->fapi.ctx = fctx;
+    return CKR_OK;
+}
+
+void backend_fapi_ctx_free(token *t) {
+    (void)(t);
+}
+
+
+#define PREFIX "/HS/SRK/tpm2-pkcs11-token-"
+
+static char * tss_path_from_id(unsigned id, const char *type) {
```

Comment:
do you modify these afterwards? const char * seems better.

Also, * tight to the variable like what you're doing on line 45.

Reply:
@williamcroberts I don't think I understand this comment ? are you referring to the return value ?

Reply:
Because I don't want to prevent modification of the returned value.